src/bson_fwd.h

```diff
@@ -0,0 +1,75 @@
+#ifndef BSON_FWD_H
+#define BSON_FWD_H
```

Comment:
Maybe `bson_common.h` would be a better name? Otherwise, the change looks good

Reply:
@jacobkeeler fixed in 
98d6cd6